tsl/src/compression/api.c

```diff
@@ -415,10 +374,7 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 	/* acquire locks on src and compress hypertable and src chunk */
 	LockRelationOid(cxt.srcht->main_table_relid, AccessShareLock);
 	LockRelationOid(cxt.compress_ht->main_table_relid, AccessShareLock);
-	LockRelationOid(cxt.srcht_chunk->table_id, ShareLock);
-
-	/* Disabling autovacuum on chunk which should be empty while in compressed state */
-	disable_autovacuum_on_chunk(chunk_relid);
+	LockRelationOid(cxt.srcht_chunk->table_id, ExclusiveLock);
```

Comment:
Why do we need to change the lock type here?

Reply:
To avoid the side effect of removing disable_autovacuum_on_chunk. Test "compression_chunk_race" was failing. After debugging figured out that SharedLock leads to deadlock during parallel compression job. If we see decompression code this table has ExclusiveLock. Previously, disable_autovacuum_on_chunk was taking RowExclusiveLock during update, unknowingly preventing the deadlock.

Reply:
The ALTER TABLE command that disabled autovacuum had an implicit ExclusiveLock